datafusion/core/tests/sql/subqueries.rs

```diff
@@ -116,3 +116,69 @@ where o_orderstatus in (
 
     Ok(())
 }
+
+#[tokio::test]
+async fn exists_subquery_with_same_table() -> Result<()> {
+    let test_repartition_joins = vec![true, false];
+    for repartition_joins in test_repartition_joins {
+        let ctx = create_join_context("t1_id", "t2_id", repartition_joins)?;
+
+        // this query will not be rewritten to join
```

Comment:
It would be better that add the reason of it.

Reply:
Added the comment.